### PR TITLE
Implement IStorageProvider OpenFile/OpenFolder/SaveFile and MessageBox dialogs

### PR DESCRIPTION
**Pull request by tomlm**
This is fix for #149 
Features:
* Implements IStorageProvider/IStorageFile/IStorageFolder
* Implements FileOpen/FolderOpen/FileSave dialogs

## File open
FileOpen

## File Open with multiple selection
FileOpenMulti

## File Save
FileSave

## Folder Select
SelectFolder

## Folder Select Multiple:
SelectFolderMulti

## MessageBox
MessageBox

# Changes
* Implements IStorageXXX interfaces as ConsoloniaStorageProvider
* Changes consolewindow to register ConsoloniaStorageProvider
* Move DialogHelper classes to Core library in Controls. This is because there is no way for an external assembly to register for IStorage, and if IStorageProvider is in Core library it can't have dependency on Turbo library
* Created **Controls** folder and **Controls/Dialog** folder
* Moved **DialogWrap/Dialog** helper classes into **Controls/Dialogs*** Moved **DialogWindow** into **Controls**, this makes DialogWindow a core control for Consolonia
* Added **FileOpenPicker** dialog with view models
* Added **FileSavePicker** Dialog with view models
* Added **FolderPicker** dialog with view models
* Added **GalleryStorage** to illustrate calling IStorageProvider to show FileOpen/FileSave/SelectFolder
* Cleaned up DialogWindow template to support Icon as a string Emoticon, cleaned up layout of DialogWindow a bit.
* Added unit tests for StorageProvider/StorageFile/StorageFolder
  
> NOTE: The icon rendering display issues are already is fixed with PR #150 

## Release Notes

- **New Features**
  - Introduced `FileOpenPicker`, `FileSavePicker`, and `FolderPicker` controls for enhanced file and folder selection.
  - Added a customizable `MessageBox` control for user interactions.
  - Implemented `GalleryStorage` and `GalleryMessageBox` for managing file operations in the gallery.

- **Improvements**
  - Enhanced `DialogWindow` and `DialogHost` functionality for better dialog management.
  - Updated `DialogWindow` properties for improved configuration options.

- **Bug Fixes**
  - Resolved issues with dialog visibility and interaction.

- **Refactor**
  - Streamlined initialization processes by removing redundant `InitializeComponent` methods across various controls.
  - Improved code clarity by directly accessing control properties instead of using `FindControl`.

- **Documentation**
  - Updated project files and configurations for better organization and clarity.

### CODE REVIEW

**Comment by tomlm** [comment]
@coderabbit please review

File: src/Consolonia.Core/Controls/DialogWindow.cs [hunk 1]
```diff
@@ -83,31 +131,60 @@ protected override Size ArrangeOverride(Size finalSize)
...
 
-        private void ShowDialogInternal(Visual parent)
+        protected void ShowDialogInternal(Visual parent)
         {
+            if (WindowStartupLocation == WindowStartupLocation.CenterScreen)
+            {
+                Width = (ushort)(parent.Bounds.Width * .9);
+                Height = (ushort)(parent.Bounds.Height * .9);
+                if (parent is Window window) window.SizeChanged += Window_SizeChanged;
+            }
```
[the remark this answers is not in the document]
Comment (tomlm) [reply src/Consolonia.Core/Controls/DialogWindow.cs:141]: we actually want these values to be ushort

File: src/Consolonia.Core/Controls/DialogWindow.cs [hunk 2]
```diff
@@ -83,31 +131,60 @@ protected override Size ArrangeOverride(Size finalSize)
...
+
             DialogHost dialogHost = GetDialogHost(parent);
             dialogHost.OpenInternal(this);
         }
 
+        private void Window_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            var window = sender as Window;
```
Comment (jinek) [inline src/Consolonia.Core/Controls/DialogWindow.cs:149]: I think sender is not supposed to be null?

File: src/Consolonia.Core/Controls/PickerViewModelBase.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            {
+                case nameof(CurrentFolderPath):
+                    try
+                    {
+                        CurrentFolder = new SystemStorageFolder(new DirectoryInfo(CurrentFolderPath));
+                    }
+                    catch (DirectoryNotFoundException)
+                    {
+                        // ignore
+                    }
```
[the remark this answers is not in the document]
Comment (jinek) [reply src/Consolonia.Core/Controls/PickerViewModelBase.cs:47]: Probably a good suggestion to catch also UnauthorizedAccessException and IOException

File: src/Tests/Consolonia.Gallery.Tests/Consolonia.Gallery.Tests.csproj [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
     <PropertyGroup>
         <IsPackable>false</IsPackable>
+        <NoWarn>CA2007;</NoWarn>
```
Comment (jinek) [inline src/Tests/Consolonia.Gallery.Tests/Consolonia.Gallery.Tests.csproj:5]: Lets suppress it for all projects?

**Comment by jinek** [comment]
I'm proposing this combination: https://github.com/jinek/Consolonia/pull/154
@tomlm Feel free to merge please

**Comment by jinek** [comment]
Proposal regarding CA2007 to disable via editorconfig https://github.com/jinek/Consolonia/pull/155

**Review by jinek — APPROVED** [review]

**Pull request merged**